tests/common/devices/sonic.py

```diff
@@ -2137,6 +2137,25 @@ def get_port_channel_status(self, port_channel_name):
         json_info = json.loads(commond_output["stdout"])
         return json_info
 
+    def get_port_fec(self, portname):
+        out = self.shell('redis-cli -n 4 HGET "PORT|{}" "fec"'.format(portname))
+        return out["stdout_lines"][0]
+
+    def set_port_fec(self, portname, state):
+        if not state:
+            state = 'none'
+        self.shell('sudo config interface fec {} {}'.format(portname, state))
+
+    def count_portlanes(self, portname):
+        out = self.shell('redis-cli -n 4 HGET "PORT|{}" "lanes"'.format(portname))
+        lanes = out["stdout_lines"][0].split(',')
```

Comment:
@antonptashnik If the output is something error when running redis-cli, such as timeout, should we have exception handle here?